app/forms/providers/means/regular_outgoings_form.rb

```diff
@@ -0,0 +1,161 @@
+module Providers
+  module Means
+    class RegularOutgoingsForm
+      include ActiveModel::Model
+
+      OUTGOING_TYPES = %w[rent_or_mortgage child_care maintenance_out legal_aid].freeze
+
+      attr_reader :transaction_type_ids, :legal_aid_application
+
+      OUTGOING_TYPES.each do |outgoing_type|
+        attr_accessor "#{outgoing_type}_amount".to_sym, "#{outgoing_type}_frequency".to_sym
+      end
+
+      validates :transaction_type_ids, presence: true, unless: :none_selected?
+      validate :all_regular_transactions_valid
+
+      def initialize(params = {})
+        @none_selected = none_selected.in?(params["transaction_type_ids"] || [])
+        @legal_aid_application = params.delete(:legal_aid_application)
+        @transaction_type_ids = params["transaction_type_ids"] || @legal_aid_application.transaction_type_ids
+
+        assign_regular_transaction_attributes
+
+        super
+      end
+
+      def save
+        return false unless valid?
+
+        ApplicationRecord.transaction do
+          destroy_transactions!
+
+          build_legal_aid_application_transaction_types
+          legal_aid_application.assign_attributes(
+            no_debit_transaction_types_selected: none_selected?,
+          )
+          legal_aid_application.save!
+
+          regular_transactions.each(&:save!)
+        end
+
+        true
+      end
+
+      def transaction_type_ids=(ids)
+        @transaction_type_ids = if none_selected?
+                                  []
+                                else
+                                  ids.compact_blank
+                                end
+      end
+
+      def transaction_type_options
+        TransactionType.not_children.debits
+      end
+
+      def none_selected
+        "none"
+      end
+
+    private
+
+      def assign_regular_transaction_attributes
+        regular_transactions.each do |transaction|
+          transaction_type = transaction.transaction_type
+          public_send("#{transaction_type.name}_amount=", transaction.amount)
+          public_send("#{transaction_type.name}_frequency=", transaction.frequency)
+        end
+      end
+
+      def none_selected?
+        @none_selected
+      end
+
+      def transaction_types
+        transaction_type_options.where(id: transaction_type_ids)
+      end
+
+      def destroy_transactions!
+        destroy_legal_aid_application_transaction_types!
+        destroy_regular_outgoing_transactions!
+        destroy_cash_outgoing_transactions!
+      end
+
+      def destroy_legal_aid_application_transaction_types!
+        legal_aid_application
+          .legal_aid_application_transaction_types
+          .debits
+          .where.not(transaction_type_id: transaction_type_ids)
+          .destroy_all
+      end
+
+      def destroy_regular_outgoing_transactions!
+        legal_aid_application
+          .regular_outgoings
+          .without(existing_outgoing_regular_transactions)
```

Comment:
I realize this copies the pattern that went in for regular incomes, so maybe @cpjmcquillan can comment, but 
is this the `Array#without` method? in which case it seem to be mixing an array method with active an record chain plus a separate query. would not the same as for cash_transactions work and be more efficient
```
.where.not(transaction_type_id: transaction_type_ids)
```

Reply:
It's an [alias for `.excluding` in ActiveRecord](https://api.rubyonrails.org/classes/ActiveRecord/QueryMethods.html#method-i-excluding), so there is no difference between the SQL generated by either statement.

If `.without(existing_outgoing_regular_transactions)` is too verbose though, then let's change it.